### PR TITLE
Add docker image name with empty spaces scenario

### PR DESCRIPTION
**Pull request by pjgg**
issue Ref: https://github.com/quarkusio/quarkus/pull/19958

A Quarkus app that uses the extension `container-image-docker` fails if the docker image name (`quarkus.application.name`) contains some spaces at the end. 

The issue has been fixed.  Verified by this PR.

### CODE REVIEW

File: docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.junit.jupiter.api.Test;
+
+import com.github.dockerjava.api.model.Image;
+
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.scenarios.annotations.DisabledOnNative;
+import io.quarkus.test.utils.DockerUtils;
+
+@QuarkusScenario
+@DisabledOnNative // in order to avoid extra test time. In my opinion native mode doesn't add value for this scenario
```
Comment (fedinskiy) [inline docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java:15]: If the test is disabled on Native, do we still need https://github.com/quarkus-qe/quarkus-test-suite/pull/438/files#diff-873490bdd3f9d04392641dd37b052242fb566f181517f6f5fabeabdce22e85a7 ?

File: docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+public class DockerBuildIT {
+
+    private static final String SPACE = "      ";
```
Comment (fedinskiy) [inline docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java:18]: I suggest to create a check for a tabulation('\t') symbol

File: docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private static final String DOCKER_IMG_VERSION = "1.0.0";
+
+    @AfterAll
+    public static void tearDown() {
+        DockerUtils.removeImage(PADDED_DOCKER_IMG_NAME, DOCKER_IMG_VERSION);
+    }
+
+    @Test
+    public void buildDockerImage() {
+        Image dockerImg = DockerUtils.getImage("qe/hello-world-app", "1.0.0");
```
Comment (fedinskiy) [inline docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java:29]: I suggest to also check for PADDED_<...>_NAME. And definitely to use DOCKER_IMG_VERSION.

File: README.md [hunk 4]
```diff
@@ -385,6 +385,9 @@ Tests:
 Checks that the application can read configuration from a ConfigMap and a Secret.
 The ConfigMap/Secret is exposed by mounting it into the container file system or the Kubernetes API server.
 
+### `docker-build`
+Checks that an application can generate a Docker image based in some configuration parameters.
```
Comment (fedinskiy) [inline README.md:389]: I presume that "in" is a typo, and it should be "on" instead.

File: docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        assertFalse(dockerImg.getId().isEmpty(), "Docker images was not created");
```
Comment (fedinskiy) [inline docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java:30]: I can see only one image being created. Am I missing something?

**Review by fedinskiy — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (pjgg) [reply README.md:389]: these files are generated by `mvn io.quarkus.platform:quarkus-maven-plugin:2.6.1.Final:create` command by default. It's true that we are not going to use it, but I prefer to leave it as it is.

File: docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java (hunk 1 above)
Comment (pjgg) [reply docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java:15]: these files are generated by `mvn io.quarkus.platform:quarkus-maven-plugin:2.6.1.Final:create` command by default. It's true that we are not going to use it, but I prefer to leave it as it is.

File: docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java (hunk 5 above)
Comment (pjgg) [reply docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java:30]: No, is fine. We are only creating a single image `qe/hello-world-app:1.0.0`, which also is removed at the end of the test.

Comment (fedinskiy) [reply docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java:31]: In that case it's probably "image" and not "images" :)

**Review by fedinskiy — CHANGES_REQUESTED** [review]
... and the tabulation tests was not commited

**Comment by fedinskiy** [comment]
@pjgg you check the tabulation(and I have doubts, that it needs to be shielded in this case) but do not check spaces. Are you sure about that?

**Comment by pjgg** [comment]
> @pjgg you check the tabulation(and I have doubts, that it needs to be shielded in this case) but do not check spaces. Are you sure about that?

Tabulation and space on the test are not required anymore in order to reproduce the issue. 

In order to reproduce you can downgrade the Quarkus version to 2.0.0.Final and run a `mvn clean verify`

**Review by fedinskiy — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -182,8 +182,9 @@ jobs:
         run: |
           if [[ -n ${MODULES_ARG} ]]; then
             echo "Running modules: ${MODULES_ARG}"
-            mvn -fae -V -B -s .github/mvn-settings.xml -fae -Dall-modules -pl $MODULES_ARG clean verify -Dnative \
-                        -Dinclude.quarkus-cli-tests -Dts.quarkus.cli.cmd="${PWD}/quarkus-dev-cli"
+            mvn -fae -V -B -s .github/mvn-settings.xml -fae -Dall-modules -Dquarkus.container-image.build=false \
```
Comment (rsvoboda) [inline .github/workflows/ci.yml:185]: Why is `-Dquarkus.container-image.build=false` needed to be added to the GH actions here and in the other places?

File: docker-build/src/main/docker/Dockerfile.legacy-jar [hunk 7]
```diff
@@ -0,0 +1,37 @@
+####
```
Comment (rsvoboda) [inline docker-build/src/main/docker/Dockerfile.legacy-jar:1]: Why is this file added when not used?

File: docker-build/src/main/docker/Dockerfile.native [hunk 8]
```diff
@@ -0,0 +1,27 @@
+####
```
Comment (rsvoboda) [inline docker-build/src/main/docker/Dockerfile.native:1]: Why is this file added when not used?

File: docker-build/src/main/docker/Dockerfile.native-distroless [hunk 9]
```diff
@@ -0,0 +1,23 @@
+####
```
Comment (rsvoboda) [inline docker-build/src/main/docker/Dockerfile.native-distroless:1]: Why is this file added when not used?

File: docker-build/src/main/resources/application.properties [hunk 10]
```diff
@@ -0,0 +1,5 @@
+quarkus.container-image.build=true
```
Comment (rsvoboda) [inline docker-build/src/main/resources/application.properties:1]: BTW, in GH action this get overridden to false

File: .github/workflows/ci.yml (hunk 6 above)
Comment (rsvoboda) [reply .github/workflows/ci.yml:185]: Comment above the `mvn` with some explanation should be present?

Is it because GH Action runner is container and you can't invoke docker build from it?

File: docker-build/src/main/docker/Dockerfile.native (hunk 8 above)
Comment (rsvoboda) [reply docker-build/src/main/docker/Dockerfile.native:1]: Seems it's used by `NativeConfigurationIT` added after this module. So I guess this file can stay.

File: docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+import org.junit.jupiter.api.Test;
+
+import com.github.dockerjava.api.model.Image;
+
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.scenarios.annotations.DisabledOnNative;
+import io.quarkus.test.utils.DockerUtils;
+
+@QuarkusScenario
+@DisabledOnNative(reason = "Due to reduce native execution time")
```
Comment (rsvoboda) [inline docker-build/src/test/java/io/quarkus/ts/docker/DockerBuildIT.java:17]: `NativeConfigurationIT` is executed in native, the native binary gets produced in native run of the TS, probably no harm to have the test enabled.

File: .github/workflows/ci.yml (hunk 6 above)
Comment (rsvoboda) [reply .github/workflows/ci.yml:185]: If this is about windows not having containers available, the property should be set in pom.xml of the module, there is already profile for Windows
